### ISSUE TITLE
[REQ][JAVA] Support method overloading

### ISSUE THREAD

**Issue by alexibz** [OpenAPITools/openapi-generator#6807]
### Is your feature request related to a problem? Please describe.

A lot of parameters in requests bodies can be optional. Currently, the Java generators generate method signatures with all of the parameters in the parameter list, requiring the caller to `null` unneeded parameters. This is fine for a handful of parameters, but sometimes there could be tens of them, making this very tedious. 
```java
AppListResponse listResponse = apiInstance.listApps(null, null);
```

## Describe the solution you'd like

Support for method overloading. So a method would have different signatures available to call. In the above case, where there are two optional parameters, a caller would be able to call:

```java
apiInstance.listApps();
apiInstance.listApps(null);
apiInstance.listApps(null, null);
```

## Describe alternatives you've considered

Another alternative would be to have parameter bags or classes for providing the parameters.

**Comment by bradvido** [comment]
Yes please! Our calls often have 20+ params, and really only the first few are required, so it's really messy and hard to read when we have to have a bunch of nulls to make the call.

**Comment by wing328** [comment]
Please try the vendor extension `x-group-parameters` with Java client generator (okhttp-gson, jersey2, native).

**Comment by bradvido** [comment]
@wing328 that looks great!  But after searching, I can't find an example of how it's used. Could you provide a quick code sample?

**Comment by madorb** [comment]
This is, IMO, a bigger problem than having to pass too many nulls.  

It's a binary compatibility problem in that it's very easy to add new _non-breaking_ (from an API perspective) optional query parameters... but this results in a _breaking_ binary change. 

Really affects the ability to evolve APIs when your client unnecessarily breaks consumers when they upgrade versions

**Comment by Hollerweger** [comment]
> This is, IMO, a bigger problem than having to pass too many nulls.
> 
> It's a binary compatibility problem in that it's very easy to add new _non-breaking_ (from an API perspective) optional query parameters... but this results in a _breaking_ binary change.
> 
> Really affects the ability to evolve APIs when your client unnecessarily breaks consumers when they upgrade versions

I don't see here any breaking change as existing api with all parameters will still exist.
Would be really great to have such a method overloading available.

Maybe as a first step on additional methods would be fine:
- Existing Method with all parameters as it is generated now
- Second one only with required parameters

For a more flexible solution a suggestion could be to allow a third overload which has only a single api parameter object which can be created via a builder.

**Comment by Hollerweger** [comment]
I found the paramter useSingleRequestParameter but it seems to only support  jersey2, jersey3, okhttp-gson
It would be great if it would be an additional overloaded method and could be supported also by others. Eg.: I'm using resttemplate library.

**Comment by mar1u50** [comment]
useSingleRequestParameter is not an option for us because it will change the whole SDK we already distributed and all the customers using the SDK will have to change their applications.

Is there any way to make the openapi generator generate all the overloads with all different parameter combinations? I'd be happy to create a PR if anyone can provide a bit of guideline/direction.

**Comment by jfiala** [comment]
I tried to generate additional operations with no parameters (thus using the default values for optional parameters) by adding additional operations in the DefaultGenerator.java, method processOperations, and it worked nicely.

However, I'm not sure if this is actually a good place to duplicate the operations.
Please also note that you need to mantain that in your custom codebase.

You can try like that:

``` java
    private OperationsMap processOperations(CodegenConfig config, String tag, List<CodegenOperation> ops, List<ModelMap> allModels) {
    	LOGGER.info("*** processOperations");
        OperationsMap operations = new OperationsMap();
        OperationMap objs = new OperationMap();
        objs.setClassname(config.toApiName(tag));
        objs.setPathPrefix(config.toApiVarName(tag));
        
        CodegenOperation newOp = new CodegenOperation();

        // TODO: Is this a good place to add additional operations to ops-List?
        for (CodegenOperation op : ops) {
        	
        	LOGGER.info("*** Operation: " + op.operationId);
        	// Create a duplicate for operationId getXXX:
        	if (op.operationId.equals("getXXX")) {
        		System.out.println("*** Details of getXXX");
        		
        		System.out.println(op);
        		
// maybe there is a more comfortable option like newOp.clone()?
        		newOp = new CodegenOperation();
        		newOp.operationId = op.operationId + "_findAll";

        		// copy properties we need of CodegenOperations 
        		newOp.authMethods = op.authMethods;
        		newOp.httpMethod = op.httpMethod;
        		newOp.path = op.path;
        		newOp.tags = op.tags;
        		newOp.baseName = op.baseName;
        		newOp.returnType = op.returnType;
...
```
